api/src/main/java/org/openmrs/CohortMembership.java

```diff
@@ -43,8 +47,18 @@ public CohortMembership(Patient patient) {
 		this(patient, new Date());
 	}
 	
-	public boolean isMemberActive() {
-		return this.getStartDate() != null && this.getStartDate().before(new Date()) && this.getEndDate() == null;
+	public boolean isMembershipActive() {
+		return this.getStartDate() != null &&
+				this.getStartDate().before(new Date()) &&
+				this.getEndDate() == null &&
+				!this.getVoided();
+	}
+	
+	public boolean isMembershipActive(Date asOfDate) {
+		return this.getStartDate() != null &&
+				!this.getStartDate().after(asOfDate) &&
```

Comment:
In fact OpenmrsUtil has utility functions you can call that compare dates taking into account null values and how you wish to treat them